txscript/standard.go

```diff
@@ -518,11 +514,10 @@ func isNullDataScript(scriptVersion uint16, script []byte) bool {
 		return true
 	}
 
-	// OP_RETURN followed by data push up to MaxDataCarrierSize bytes.
+	// OP_RETURN followed by a data push.
 	tokenizer := MakeScriptTokenizer(scriptVersion, script[1:])
 	return tokenizer.Next() && tokenizer.Done() &&
-		(IsSmallInt(tokenizer.Opcode()) || tokenizer.Opcode() <= OP_PUSHDATA4) &&
-		len(tokenizer.Data()) <= MaxDataCarrierSize
+		(IsSmallInt(tokenizer.Opcode()) || tokenizer.Opcode() <= OP_PUSHDATA4)
```

Comment:
Not familiar with btcd's code but this appears to mandate a single data push to follow the `OP_RETURN` for it to be standard. This is not Bitcoin Core's behaviour. If you want to match the behaviour from https://github.com/bitcoin/bitcoin/pull/32406 then i think just unconditionally `return`ing `len(script) >= 1 && script[0] == OP_RETURN` on line 508 above is all you need.

Reply:
FWIW, we haven't kept up with `bitcoind` discretionary policy for quite some time now (many years). There're no BIPS for policies like this, so it makes it hard to follow, and if it doesn't need interop, then there isn't a strong reason to copy behavior. There's a PR open to update it to allow multiple pushes. But in LN, with the new rbf close flow, `OP_RETURN` is defined as it is here: the more restrictive version.